Octree/include/CGAL/Octree.h

```diff
@@ -0,0 +1,646 @@
+
+    // Shift the squared box to make sure it's centered in the original place
+    Point bbox_transformed_centroid = midpoint(bbox.min(), bbox.max());
+    Vector diff_centroid = bbox_centroid - bbox_transformed_centroid;
+    bbox = bbox.transform(Aff_transformation_3<Kernel>(TRANSLATION, diff_centroid));
+
+    // save octree attributes
+    m_bbox_min = bbox.min();
+    m_bbox_side = bbox.max()[0] - m_bbox_min[0];
+    m_root.points() = {point_range.begin(), point_range.end()};
+
+  }
+
+  /// @}
+
+  /// \name Tree Building
+  /// @{
+
+  /*!
+   * \brief Subdivide an octree's nodes and sub-nodes until it meets the given criteria
+   *
+   * The split criterion can be any function pointer that takes a Node pointer
+   * and returns a boolean value (where true implies that a Node needs to be split).
+   * It's safe to call this function repeatedly, and with different criterion.
+   *
+   * \param split_criterion rule to use when determining whether or not a node needs to be subdivided
+   */
+  void refine(const Split_criterion_function &split_criterion) {
+
+    // If the tree has already been refined, reset it
+    if (!m_root.is_leaf())
+      m_root.unsplit();
+
+    // Make sure the max depth is reset as well
+    m_max_depth_reached = 0;
+
+    // create a side length map
+    for (int i = 0; i <= (int) 32; i++)
+      m_side_per_depth.push_back(m_bbox_side / (FT) (1 << i));
+
+    // Initialize a queue of nodes that need to be refined
+    std::queue<Node *> todo;
+    todo.push(&m_root);
+
+    // Process items in the queue until it's consumed fully
+    while (!todo.empty()) {
+
+      // Get the next element
+      auto current = todo.front();
+      todo.pop();
+      int depth = current->depth();
+
+      // Check if this node needs to be processed
+      if (split_criterion(*current)) {
+
+        // Split the node, redistributing its points to its children
+        split((*current));
+
+        // Process each of its children
+        for (int i = 0; i < 8; ++i)
+          todo.push(&(*current)[i]);
+
+      }
+    }
+  }
+
+  /*!
+   * \brief Refine an octree using a max depth and max number of points in a node as split criterion
+   *
+   * This is equivalent to calling:
+   *
+   *     refine(Split_criterion::Max_depth_or_bucket_size(max_depth, bucket_size));
+   *
+   * This functionality is provided for consistency with older octree implementations
+   * which did not allow for custom split criterion.
+   *
+   * \param max_depth deepest a tree is allowed to be (nodes at this depth will not be split)
+   * \param bucket_size maximum points a node is allowed to contain
+   */
+  void refine(size_t max_depth = 10, size_t bucket_size = 20) {
+    refine(Split_criterion::Max_depth_or_bucket_size(max_depth, bucket_size));
+  }
+
+  /// @}
+
+  /// \name Accessors
+  /// @{
+
+  /*!
+   * \brief Provides read-only access to the root node, and by extension the rest of the tree
```

Comment:
It's a non-written convention to start with lower case so that the function signature and the description together form a sentence.

Reply:
Thanks for the tip!